add js package

GSemikozov opened the pull request:
# genai-prices package (JS/TS)

## Testing

### Node.js Test

Run:

```bash
node tests/test-error-handling.js
```

This tests error handling, sync/async API, and providerId usage.

### Browser Test

1. Build the package: `npm run build`
2. Serve the directory: `npx serve .` or `python3 -m http.server`
3. Open `tests/test-browser.html` in your browser.
4. Enter a provider (e.g., `openai`) and model (e.g., `gpt-3.5-turbo`) and run the test.

### Global CLI Installation

You can install the CLI globally to use the `genai-prices` command from anywhere:

```bash
npm install -g @pydantic/genai-prices
```

After installing globally, you can run:

```bash
genai-prices calc gpt-4 --input-tokens 1000 --output-tokens 500
genai-prices list
```

### CLI

After global installation, you can use the CLI as follows:

```bash
# Basic usage
genai-prices gpt-3.5-turbo --input-tokens 1000 --output-tokens 100

# With auto-update (fetches latest prices from GitHub)
genai-prices gpt-3.5-turbo --input-tokens 1000 --output-tokens 100 --auto-update

# Specify provider explicitly
genai-prices openai:gpt-3.5-turbo --input-tokens 1000 --output-tokens 100

# List available providers and models
genai-prices list
genai-prices list openai
```

GSemikozov commented on packages/js/package.json:
```diff
@@ -0,0 +1,59 @@
+  },
+  "scripts": {
+    "build": "vite build && npm run build:cli",
+    "build:cli": "tsc src/cli.ts --outDir dist --module nodenext --esModuleInterop --skipLibCheck",
```
yeah, I can use additional plugin and multi-entries flow, it is possible. But cli with tsc is more common way I guess.

samuelcolvin commented on packages/js/README.md:
```diff
@@ -0,0 +1,102 @@
+
+## CLI Usage
+
+### List all providers and models
```
We can, but we don't need two clis if it's more work.

GSemikozov replied:
I updated it, we can use both.

samuelcolvin commented on packages/js/.gitignore:
you shouldn't need this, the root `.gitignore` should cover it.

samuelcolvin reviewed (commented):
I haven't reviewed this fully, looks good over all.

But most importantly, we need to add:
* `data.ts` - the data itself, you can basically build this with a template that's built from `export data: Providers = {copy_data_as_json}` then run prettier over it
* remove any references to node types so it runs in the browser
* test it somehow in the browser

GSemikozov commented on packages/js/src/dataLoader.ts:
```diff
@@ -0,0 +1,197 @@
+import fs from 'fs'
```
update solution, now we can use this package in any environment (browser, node, cloudflare, deno, etc)

GSemikozov commented:
> I haven't reviewed this fully, looks good over all.
> 
> But most importantly, we need to add:
> 
> * `data.ts` - the data itself, you can basically build this with a template that's built from `export data: Providers = {copy_data_as_json}` then run prettier over it
> * remove any references to node types so it runs in the browser
> * test it somehow in the browser

Added in separate PR (already merged). Now we can use package in any env, including browser, cloudflare on node. Added tests for browser as well (see PR description).
